projects/npm-tools/packages/npm-scripts/src/jsp/checkExposesConfiguration.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/* eslint-disable no-console */
+
+const fs = require('fs');
+const path = require('path');
+
+const getMergedConfig = require('../utils/getMergedConfig');
+const getTaglibInvocations = require('./getTaglibInvocations');
+
+const TAGLIBS = {
+	'aui:script': {
+		attributes: ['require'],
+	},
+	'liferay-frontend:component': {
+		attributes: ['module'],
+	},
+	'react:component': {
+		attributes: ['module', 'propsTransformer'],
+	},
+	'soy:template-renderer': {
+		attributes: ['module'],
+	},
+};
+
+/**
+ * Scans JSP files for JS modules and reports linting errors for those not
+ * included in the `exposes` section for the `npmscripts.config.js` file.
+ */
+module.exports = function (source) {
+	const config = getMergedConfig('npmscripts');
+
+	if (!fs.existsSync(config.build.input)) {
+		return [];
+	}
+
+	const invocations = getTaglibInvocations(source, ...Object.keys(TAGLIBS));
+
+	const moduleNames = new Set();
+
+	for (const invocation of invocations) {
+		const {attributes: validAttributes} = TAGLIBS[invocation.tag];
+		const {attributes} = invocation;
+
+		for (const validAttribute of validAttributes) {
+			const require = attributes[validAttribute];
```

Comment:
Shadowing `require` here is a bit 🤮  — can you rename this? Maybe just `attribute`?